docs/site/tutorials/protocol_oriented_generics.ipynb

```diff
@@ -0,0 +1,1108 @@
+{
```

Comment:
<p><strong>Brennan: </strong></p><blockquote>"Hmm, this section reads both relatively dry and potentially leaves people getting hung up on Swift limitations (e.g. that C++ doesn't have).&nbsp;</blockquote><blockquote><br></blockquote><blockquote>What do you think about this silly idea: what if you illustrated things by example such that the user themselves comes to the conclusion that subclassing / object orientation is inappropriate for certain situations? You'll definitely have to figure out a great example, but I think showing (not telling) might be more powerful here. Does that make sense?"</blockquote><p><br></p><p><strong>Bart:</strong></p><blockquote>"Hmm I'll definitely try figuring something out! My worry is that I may introduce some bias and throw object oriented under the bus, i.e protocol oriented is superior and object oriented sucks, which my previous version of the tutorial sometimes did and Eugene&nbsp;helped correct."</blockquote>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/swift/pull/187/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>